pkgs/development/libraries/fpattern/default.nix

```
@@ -0,0 +1,29 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation (finalAttrs: {
+  version = "1.9";
+  pname = "fpattern";
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "Loadmaster";
+    repo = finalAttrs.pname;
```

```suggestion
    repo = "fpattern";
```
this is actually wrong because a change in pname is most likely not suppose to trigger a change in the repo.

---

Agreed in principle but I must point out that half of the existing packages seem to make the same mistake. I'll avoid it in future but it doesn't seem meaningful to correct now, unless you feel very strongly.

Thanks for your attention to these PRs btw.

```
nixpkgs/pkgs on  master [!] 
$ \grep -R 'repo = "' .  | wc -l   
10806

nixpkgs/pkgs on  master [!] 
$ \grep -R 'repo = pname' .  | wc -l
4794
```

---

I disagree. The fact that half of nixpkgs is operating in a certain way does not automatically mean you should follow suit, especially when advised against by a maintainer.

What I'm suggesting is that despite any flaws present in nixpkgs, it is always more effective to make improvements at your level. This aligns with the philosophy of "thinking globally, acting locally".

I also understand that altering a system that is functioning well can be cumbersome, but as maintainers, we strive to provide the best objective guidance to contributors.

We expect that they grasp this concept.

---

I'm thankful for the guidance, but if this is the chosen pattern then it seems like a missed opportunity to bring in a linter during PR evaluation or to go further and update the existing files as seems to be occurring in PRs [such as this](https://github.com/NixOS/nixpkgs/pull/239264/files). Otherwise the volume of the bad implementations will outweigh comments such as these.

Here's the PR to fix: https://github.com/NixOS/nixpkgs/pull/241217

Thanks again to both of you for your time.